src/_pytest/python.py

```diff
@@ -871,25 +868,12 @@ def xunit_setup_method_fixture(self, request) -> Generator[None, None, None]:
         self.obj.__pytest_setup_method = xunit_setup_method_fixture
 
 
-class Instance(PyCollector):
-    _ALLOW_MARKERS = False  # hack, destroy later
-    # Instances share the object with their parents in a way
-    # that duplicates markers instances if not taken out
-    # can be removed at node structure reorganization time.
-
-    def _getobj(self):
-        # TODO: Improve the type of `parent` such that assert/ignore aren't needed.
-        assert self.parent is not None
-        obj = self.parent.obj  # type: ignore[attr-defined]
-        return obj()
-
-    def collect(self) -> Iterable[Union[nodes.Item, nodes.Collector]]:
-        self.session._fixturemanager.parsefactories(self)
-        return super().collect()
-
-    def newinstance(self):
-        self.obj = self._getobj()
-        return self.obj
+# Instance used to be a node type between Class and Function.  It has been
+# removed in pytest 7.0. Some plugins exist which reference `pytest.Instance`
+# only to ignore it; this dummy class keeps them working. This could probably
```

Comment:
It's tempting but I think it won't be very cool of us to entirely drop the name without a deprecation period, as it *is* public and being referenced, if only to ignore it.

We can show a deprecation warning when the name is imported using module `__getattr__`. It's only available since Python 3.7 but maybe that's good enough?

Reply:
> We can show a deprecation warning when the name is imported using module __getattr__. It's only available since Python 3.7 but maybe that's good enough?

Haven't thought of that.

My suggestion to just remove it was because I couldn't think of a way to deprecate it, but that solves it. 👍

Reply:
Agreed - I think showing a deprecation warning in most configurations is much better than not showing a deprecation warning at all. And after all, Python 3.6 [is end-of-life](https://devguide.python.org/#status-of-python-branches) on December 23rd, so I agree with  @nicoddemus. I'd even say that if a plugin is not tested on 3.7+ yet, it probably has bigger bitrotting problems.